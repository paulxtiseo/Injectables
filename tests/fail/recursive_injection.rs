use injectables::{injectable, inject_fields};

#[injectable]
#[inject_fields(RecursiveStruct)]  // Tries to inject its own fields
pub struct RecursiveStruct {
  pub id: u64,
}

fn main() {
  let test = RecursiveStruct {
    id: 1,
  };
}