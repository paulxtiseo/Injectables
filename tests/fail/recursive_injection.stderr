error: Recursive injection detected: RecursiveStruct tries to inject from itself
 --> tests/fail/recursive_injection.rs:4:1
  |
4 | #[inject_fields(RecursiveStruct)]  // Tries to inject its own fields
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0422]: cannot find struct, variant or union type `RecursiveStruct` in this scope
  --> tests/fail/recursive_injection.rs:10:14
   |
10 |   let test = RecursiveStruct {
   |              ^^^^^^^^^^^^^^^ not found in this scope
